tests/units/plugins/filter/test_merge.py

```diff
@@ -0,0 +1,258 @@
+"""
+Tests for the "merge" plugin filter
+"""
+from copy import deepcopy
+
+from ansible.compat.tests import unittest
+from ansible.errors import AnsibleFilterError
+
+from plugins.filter.merge import merge_with_app
+
+
+def deep_sort_dict(d):
+    """
+        Recursively alphabetically sort lists as dictionary values (input dict
+        is not modified)
+    """
+    for k, v in d.iteritems():
+        if isinstance(v, list):
+            for i in v:
+                if isinstance(i, dict):
+                    deep_sort_dict(i)
+            d[k] = sorted(v)
+        if isinstance(v, dict):
+            deep_sort_dict(v)
+    return d
+
+
+class TestDeepSortDict(unittest.TestCase):
+    """Test the ``deep_sort_dict`` utility"""
+
+    def test_flat_dict(self):
+        """Avoid testing recusive calls"""
```

Comment:
s/recusive/recursive
Actually I don't understand how this docstring describes what you are testing.

Maybe something like: "The deep_sort_dict method should sort flat (non nested) dictionaries as expected" ?

Reply:
Right. I've fixed this docstring.